WalletWasabi/WabiSabi/Client/Batching/PaymentBatch.cs

```diff
@@ -21,20 +23,45 @@ public class PaymentBatch
 {
 	private readonly List<Payment> _payments = new();
 	private readonly object _syncObj = new();
-	private IEnumerable<PendingPayment> PendingPayments => GetPayments().OfType<PendingPayment>();
-	private IEnumerable<InProgressPayment> InProgressPayments => GetPayments().OfType<InProgressPayment>();
+	private IEnumerable<Payment> PendingPayments => GetPayments().Where(p => p.State is PendingPayment);
+	private IEnumerable<Payment> InProgressPayments => GetPayments().Where(p => p.State is InProgressPayment);
 
 	public Guid AddPayment(IDestination destination, Money amount)
 	{
-		var payment = new PendingPayment(destination, amount);
+		var payment = new Payment(destination, amount);
 		lock (_syncObj)
 		{
 			_payments.Add(payment);
 		}
-		Logger.LogInfo($"Payment for {payment.Amount} BTC to {payment.Destination.ScriptPubKey}.");
+		Logger.LogInfo($"Payment {payment.Id} for {payment.Amount} BTC to {payment.Destination.ScriptPubKey}.");
 		return payment.Id;
 	}
 
+	public void AbortPayment(Guid id)
+	{
+		lock (_syncObj)
```

Comment:
Could we avoid using lock operator via replacing it by concurrent collections?

[Ms. doc.](https://learn.microsoft.com/en-us/dotnet/api/system.collections.concurrent.blockingcollection-1?view=net-8.0)

Reply:
No need of concurrent collection here. Using that would require to make the class disposable and deal with that shit.